upflux_db.sql

```diff
@@ -3,125 +3,154 @@
 USE upflux;
 
 /*Create Table Statements*/
-CREATE TABLE Users (
-    user_id INT NOT NULL AUTO_INCREMENT,
-    name VARCHAR(100) NOT NULL,
-    email VARCHAR(255) NOT NULL,
-    password_hash VARCHAR(255) NOT NULL,
-    PRIMARY KEY (user_id)
-);
 
+-- Create Machines Table
 CREATE TABLE Machines (
-    machine_id INT NOT NULL AUTO_INCREMENT,
+    machine_id INT AUTO_INCREMENT PRIMARY KEY,
     machine_status ENUM('Alive', 'Shutdown', 'Unknown') NOT NULL,
     memory_usage FLOAT,
-    activity_status ENUM('Busy', 'Idle', 'Offline'),
-    PRIMARY KEY (machine_id)
+    activity_status ENUM('Busy', 'Idle', 'Offline') NOT NULL
 );
 
-CREATE TABLE Packages (
-    package_id INT NOT NULL AUTO_INCREMENT,
-    version_number FLOAT NOT NULL,
-    package_size FLOAT NOT NULL,
-    package_signature VARCHAR(255),
-    release_date TIMESTAMP NOT NULL,
-    PRIMARY KEY (package_id)
+-- Create Users Table
+CREATE TABLE Users (
+    user_id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    email VARCHAR(255) NOT NULL,
+    role ENUM('Admin', 'Engineer') NOT NULL
+);
+
+-- Create Admin_Details Table 
+CREATE TABLE Admin_Details (
+    admin_id INT AUTO_INCREMENT PRIMARY KEY,
+    user_id INT NOT NULL,
+    password_hash VARCHAR(255) NOT NULL,
+    FOREIGN KEY (user_id) REFERENCES Users(user_id)
 );
 
-CREATE TABLE Licences (
-    licence_key VARCHAR(255) NOT NULL,
+-- Create Licenses Table
+CREATE TABLE Licenses (
+    licence_key VARCHAR(255) NOT NULL PRIMARY KEY,
     machine_id INT NOT NULL,
     validity_status VARCHAR(50) NOT NULL,
     expiration_date TIMESTAMP NOT NULL,
-    PRIMARY KEY (licence_key),
     FOREIGN KEY (machine_id) REFERENCES Machines(machine_id)
 );
 
+-- Create Credentials Table

```

Comment:
Like we discussed, there might be a conflict here, when a token expores and that same user gets another token with the same access to the machines. So a better PK has to be used

Reply:
Thanks for the feedback. I have now added a new unique primary key for the credentials table as well as an 'expires_at' field to track when a token expires.